### DISCUSSION TITLE
Bundle for the web with workers, pgvector and live?

### DISCUSSION THREAD

**Discussion by kikohs** [electric-sql/pglite#216] [Q&A]
Hello,

I'm trying to bundle the library for the web with webworker support and `vector` and `live`. I would like to package it into a single JS file that could be hosted on a  CDN (or at least a single js that loads the wasm, data and gz).
I can't manage to find the right webpack config for it.

`index.js`
``` js
import { PGliteWorker } from '@electric-sql/pglite/worker'
import { live } from '@electric-sql/pglite/live'

export const createWorker = async (dataDir, meta) => {
    try {
        return await PGliteWorker.create(
            new Worker(new URL('./pgworker.js', import.meta.url), {
                type: 'module',
            }),
            {
                extensions: {
                    live
                },
                dataDir,
                meta,
            },
        );
    } catch (error) {
        console.error("Error in createWorker:", error);
        throw error;
    }
};
```

`pgworker.js`
```js
import { PGlite } from '@electric-sql/pglite'
import { worker } from '@electric-sql/pglite/worker'
import { live } from '@electric-sql/pglite/live'
import { vector } from '@electric-sql/pglite/vector'

worker({
    async init(options) {
        console.log('PGlite Worker started');
        // const meta = options.meta;  
        return new PGlite({
            dataDir: options.dataDir,
            extensions: {
                vector,
                live
            },
        })
  },
})```

**Comment by samwillis** [comment]
Hey @kikohs 

I'm not sure you will be able to bundling down to a single js file as the worker will need a js fie for just itself, and ideally you would want to have some shared resources between the two. It may be possible with a little hacking though.

Can I ask why you want to only have a single js file?

**Comment by kikohs** [comment]
Thanks for your message. I'm trying to integrate PGlite into https://cables.gl. It's a node-based visual framework/editor (online) and also now desktop as an electron app. 
Loading libraries works with IIFE or a module loaded dynamically in cables. 


To be fair, I couldn't package it for the electron version as well. I get errors while loading the different chunks created by webpack. 

The subtlety is that here I build blocks for others and I don't know in advance what libraries you might put. Each block is an independent micro-api. 
We don't package everything as we would with a regular node.js project. 


In cables. I create workers dynamically by injecting the code. It's annoying but it works..

Hope it helps. 

PGlite looks amazing!
